src/domain/QuestionList/QuestionSetView.tsx

```diff
@@ -0,0 +1,77 @@
+import { LoadingOutlined } from '@ant-design/icons';
+import { Col, List, Row, Spin } from 'antd';
+import React from 'react';
+
+import { arrayDistinct } from '../../services/arrayUtils';
+import { QuestionModel, QuestionSetDetail } from '../../services/Client';
+import { average } from '../../services/mathUtils';
+import QuestionCard from '../Question/QuestionCard/QuestionCard';
+import AverageDifficultyCircle from './AverageDifficultyCircle';
+import DistinctCategoryTags from './DistinctCategoryTags';
+import styles from './QuestionSetView.module.scss';
+import QuestionSetViewHeader from './QuestionSetViewHeader';
+
+export default function QuestionSetView({
+  list,
+  removeQuestionFromListCallback,
+  updateTitleCallback,
+}: {
+  list: QuestionSetDetail | undefined;
+  removeQuestionFromListCallback: (id: number) => void;
+  updateTitleCallback: (title: string) => void;
+}): JSX.Element {
+  if (!list) {
+    return <Spin indicator={<LoadingOutlined />} />;
+  }
+
+  return (
+    <Row className={`${styles.questionSetBackground} full-height`}>
+      <Col span={1} />
+      <Col span={22}>
+        <QuestionSetViewHeader
+          title={list?.questionSet?.title ?? ''}
+          createdBy="anonymous"
+          updateTitleCallback={updateTitleCallback}
+        />
+        <div className={styles.questionSetView}>
+          <AverageDifficultyCircle
+            percent={Number(
+              (
+                average(
+                  list.questions
+                    ?.filter((question) => question.difficulty)
+                    .map((question) => question.difficulty ?? 0) ?? []
+                ) * 20
+              ).toFixed(2)
```

Comment:
I think it's not ideal to compute the average difficulty in this place.  I propose we rework this to a separate function that will take an array of questions and returns average difficulty. That would create one place responsible for such logic and with future changes to the difficulty attribute (already planned), we will have a more resilient solution. 

Regarding where should be such function stored, I would not hesitate to put it into some sort of service.